http/src/lib.nr

```diff
@@ -64,7 +61,7 @@ impl Parser {
         polynomial_digest(self.into(), polynomial_input, 1)
     }
 
-    fn update_state(self, character: u8, polynomial_input: Field) -> Self {
+    fn update_state(self, character: u8) -> Self {
```

Comment:
👍 

Okay i see what you mean now. However, I do think we want to produce a polynomial hash of:
- header names
- header values
- body
Wasn't this used for that?

Reply:
As I understand it,  for `header names, header values, for parsing body` no polynomial hash is required.
polynomial hashes is produced like `line_digests, body_digest`. And for these, we calculate it outside of FSM.

So I think `polynomial_input` is not neccesary in function update_state

Reply:
Yeah that makes sense. Hashing certainly can happen outside the machine!